llvm/include/llvm/Analysis/GenericDomTreeUpdaterImpl.h

```diff
@@ -229,57 +246,58 @@ GenericDomTreeUpdater<DerivedT, DomTreeT, PostDomTreeT>::dump() const {
   }
 
   OS << "Pending DeletedBBs:\n";
-  Index = 0;
-  for (const auto *BB : DeletedBBs) {
+  for (auto [Index, BB] : enumerate(DeletedBBs)) {
     OS << "  " << Index << " : ";
-    ++Index;
     if (BB->hasName())
       OS << BB->getName() << "(";
     else
-      OS << "(no_name)(";
+      OS << "(no name)(";
     OS << BB << ")\n";
   }
 #endif
 }
 
 template <typename DerivedT, typename DomTreeT, typename PostDomTreeT>
+template <bool IsForward>
 void GenericDomTreeUpdater<DerivedT, DomTreeT,
-                           PostDomTreeT>::applyDomTreeUpdates() {
+                           PostDomTreeT>::applyUpdatesImpl() {
+  auto *DomTree = [&]() {
+    if constexpr (IsForward)
+      return DT;
+    else
```

Comment:
Nit: no else after return

Reply:
Unfortunately the type of dominator tree and post dominator tree are different, this ensures compiler can deduce correct type.